app/reports/page-content.tsx

```diff
@@ -356,29 +357,50 @@ function useModerationQueueQuery() {
         }
       })
 
-      const { data } =
-        await labelerAgent.api.tools.ozone.moderation.queryStatuses({
-          limit: 50,
-          includeMuted: true,
-          ...queryParams,
-        })
-
-      const queueDivider = QUEUE_NAMES.length
-      const queueIndex = QUEUE_NAMES.indexOf(queueName ?? '')
-      const statusesInQueue = queueName
-        ? data.subjectStatuses.filter((status) => {
-            const subjectDid =
-              status.subject.$type === 'com.atproto.admin.defs#repoRef'
-                ? status.subject.did
-                : new AtUri(`${status.subject.uri}`).host
-            const queueDeciderCharCode =
-              `${subjectDid}`.split(':').pop()?.charCodeAt(0) || 0
-            return queueDeciderCharCode % queueDivider === queueIndex
-          })
-        : data.subjectStatuses
-
-      return { cursor: data.cursor, subjectStatuses: statusesInQueue }
+      return getQueueItems(labelerAgent, queryParams, queueName)
     },
     getNextPageParam: (lastPage) => lastPage.cursor,
   })
 }
+
+const getQueueItems = async (
+  labelerAgent: Agent,
+  queryParams: ToolsOzoneModerationQueryStatuses.QueryParams,
+  queueName: string | null,
+) => {
+  const pageSize = 50
+  const { data } = await labelerAgent.tools.ozone.moderation.queryStatuses({
+    limit: pageSize,
+    includeMuted: true,
+    ...queryParams,
+  })
+
+  const queueDivider = QUEUE_NAMES.length
+  const queueIndex = QUEUE_NAMES.indexOf(queueName ?? '')
+  const statusesInQueue = queueName
+    ? data.subjectStatuses.filter((status) => {
+        const subjectDid =
+          status.subject.$type === 'com.atproto.admin.defs#repoRef'
+            ? status.subject.did
+            : new AtUri(`${status.subject.uri}`).host
+        const queueDeciderCharCode =
+          `${subjectDid}`.split(':').pop()?.charCodeAt(0) || 0
+        return queueDeciderCharCode % queueDivider === queueIndex
+      })
+    : data.subjectStatuses
+
+  // This is a recursive call to get items in queue if the current page
+  // gives us less than full page size and there are more items to fetch
+  if (statusesInQueue.length === 0 && data.cursor) {
+    return getQueueItems(
```

Comment:
Only think I am a little unsure about here is that it would be easy to create a bug on the server side that causes this to spin and cause tons of requests.  The fix, I think, would be to pass a counter as a last arg to `getQueueItems()` that ensures we don't fetching more than maybe 10 pages this way.